core/src/test/java/com/linecorp/armeria/client/HttpClientIntegrationTest.java

```diff
@@ -724,6 +734,39 @@ void testUpgradeRequestExecutesLogicOnlyOnce() throws Exception {
         clientFactory.close();
     }
 
+    @ParameterizedTest
+    @ValueSource(booleans = { true, false })
+    void requestAbortWithException(boolean isAbort) {
+        final HttpClient client = HttpClient.of(server.httpUri("/"));
+        final HttpRequestWriter request = HttpRequest.streaming(HttpMethod.GET, "/client-aborted");
+
+        final HttpResponse response = client.execute(request);
+        final IllegalStateException badState = new IllegalStateException("bad state");
+        if (isAbort) {
+            request.abort(badState);
+        } else {
+            request.close(badState);
+        }
+        // request cause is obtained immediately
```

Comment:
What does this comment mean? I guess `join` could take time if it wanted to so we're not verifying anything immediate.

Reply:
Oops, I should've deleted this comment. There was a code that holds an exception when a stream has been aborted. Thanks for reviewing!